worker/workers/framework.py

```diff
@@ -58,9 +56,8 @@ def process_jobs(self):
         if len(self.waiting_jobs) < self.bridge_data.queue_size:
             self.add_job_to_queue()
         # Start new jobs
-        while len(self.running_jobs) < self.bridge_data.max_threads:
```

Comment:
You can leave that in for now but for readability, try to avoid adding too many conditionals. It's easier to understand with multiple if/break cases